diff --git a/tests/integration/test_authd/data/test_authd_use_source_ip.yaml b/tests/integration/test_authd/data/test_authd_use_source_ip.yaml
index 176b6e44c..ff48749b7 100644
--- a/tests/integration/test_authd/data/test_authd_use_source_ip.yaml
+++ b/tests/integration/test_authd/data/test_authd_use_source_ip.yaml
@@ -43,3 +43,15 @@
         status: 'success'
         name: 'user1'
         ip: '127.0.0.1'
+  -
+    name: 'Let manager decide IPv6'
+    description: 'Let manager decide'
+    ipv6: 'yes'
+    ip_specified: 'yes'
+    test_case:
+    -
+      input: "OSSEC A:'user1' IP:'src'"
+      output:
+        status: 'success'
+        name: 'user1'
+        ip: ' ::1/128'
diff --git a/tests/integration/test_authd/test_authd_use_source_ip.py b/tests/integration/test_authd/test_authd_use_source_ip.py
index 8d16d139b..f395052c6 100644
--- a/tests/integration/test_authd/test_authd_use_source_ip.py
+++ b/tests/integration/test_authd/test_authd_use_source_ip.py
@@ -83,7 +83,12 @@ configurations = load_wazuh_configurations(configurations_path, __name__, params
 # Variables
 
 log_monitor_paths = []
-receiver_sockets_params = [(("localhost", 1515), 'AF_INET', 'SSL_TLSv1_2')]
+for test_case in test_authd_use_source_ip_tests:
+    if 'ipv6' in test_case:
+        receiver_sockets_params = [(("localhost", 1515), 'AF_INET6', 'SSL_TLSv1_2')]
+    else:
+        receiver_sockets_params = [(("localhost", 1515), 'AF_INET', 'SSL_TLSv1_2')]
+
 monitored_sockets_params = [('wazuh-modulesd', None, True), ('wazuh-db', None, True), ('wazuh-authd', None, True)]
 receiver_sockets, monitored_sockets, log_monitors = None, None, None  # Set in the fixtures
 
